server/ai_process.go

```diff
@@ -1098,10 +1101,17 @@ func submitLiveVideoToVideo(ctx context.Context, params aiRequestParams, sess *A
 			monitor.AIFirstSegmentDelay(delayMs, sess.OrchestratorInfo)
 		}
 		clog.V(common.VERBOSE).Infof(ctx, "First Segment delay=%dms streamID=%s", delayMs, params.liveParams.streamID)
+		firstSegmentReceived <- struct{}{}
 
 	})
 	startEventsSubscribe(ctx, events, params, sess)
-	return resp, nil
+	select {
+	case <-firstSegmentReceived:
+		cancelCtx()
```

Comment:
Did you mean to cancel on the other `case` here, of timeout? IIUC this will cancel all trickle subs from the stream